Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -192,6 +193,13 @@ FullDiagnostics::InitializeFieldFunctorsRZopenPMD (int lev)
     m_all_field_functors[lev].clear();
     m_all_field_functors[lev].resize(m_varnames_fields.size());
 
+    // Boolean flag for whether the current density should be deposited before
+    // diagnostic output
+    bool deposit_current = (
```

Comment:
I can approve this as it is, but it would be nice if this bit of duplicated code could be avoided. Maybe create a class parameter like `solver_deposits_current` that is set in the constructor. Then this variable is set to `bool deposit_current = !solver_deposits_current`.

Reply:
That's a good point since if we add more solvers that do not deposit current we will now only need to update this flag in one place. Thanks for the suggestion!